plugins/verifier/cosign/cosign.go

```diff
@@ -78,84 +77,134 @@ func parseInput(stdin []byte) (*PluginInputConfig, error) {
 }
 
 func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, referenceDescriptor ocispecs.ReferenceDescriptor, referrerStore referrerstore.ReferrerStore) (*verifier.VerifierResult, error) {
+	ctx := context.Background()
 	input, err := parseInput(args.StdinData)
 	if err != nil {
 		return nil, err
 	}
+	keyRef := input.Config.KeyRef
+	rekorURL := input.Config.RekorURL
+	cosignOpts := &cosign.CheckOpts{
+		ClaimVerifier: cosign.SimpleClaimVerifier,
+	}
 
-	payload, _, err := signatures(context.Background(), subjectReference.Original, input.Config.KeyRef, input)
+	var ecdsaVerifier signature.Verifier
+	var roots *x509.CertPool
+	if keyRef != "" {
+		ecdsaVerifier, err = loadPublicKey(ctx, keyRef)
+		if err != nil {
+			return &verifier.VerifierResult{
```

Comment:
nit: not a strong opinion, I feel we can have a function that takes in an errMsg and err, and returns the failed verifierResult.

Reply:
agreed. Good idea